### PR TITLE
Add local histogram vec

### PR DESCRIPTION
**Pull request by overvenus**

### CODE REVIEW

File: src/histogram.rs [hunk 1]
```diff
@@ -710,6 +725,41 @@ impl Drop for LocalHistogram {
...
+    /// Remove a `LocalHistogram` by label values.
+    /// See more [MetricVec::remove_label_values]
+    /// (/prometheus/struct.MetricVec.html#method.remove_label_values)
+    pub fn remove_label_values(&mut self, vals: &[&str]) -> Result<()> {
+        let hash = self.vec.v.hash_label_values(vals)?;
+        self.local.remove(&hash);
+        self.vec.v.delete_label_values(vals)
+    }
+
+    pub fn flush(&mut self) {
```
Comment (siddontang) [inline src/histogram.rs]: add comment

File: src/histogram.rs [hunk 2]
```diff
@@ -710,6 +725,41 @@ impl Drop for LocalHistogram {
...
+        let vec = &self.vec;
+        self.local
+            .entry(hash)
+            .or_insert_with(|| vec.with_label_values(vals).local())
+    }
+
+    /// Remove a `LocalHistogram` by label values.
+    /// See more [MetricVec::remove_label_values]
+    /// (/prometheus/struct.MetricVec.html#method.remove_label_values)
+    pub fn remove_label_values(&mut self, vals: &[&str]) -> Result<()> {
```
Comment (siddontang) [inline src/histogram.rs]: seem no need remove here

File: src/histogram.rs (hunk 1 above)
Comment (siddontang) [reply src/histogram.rs]: you can use drain to clear all values.

**Comment by siddontang** [comment]
PTAL @disksing

File: src/histogram.rs (hunk 1 above)
Comment (overvenus) [reply src/histogram.rs]: I do not think we need drain, drain drops all cached `LocalHistogram`s.

**Comment by siddontang** [comment]
LGTM

File: src/histogram.rs [hunk 3]
```diff
@@ -710,6 +725,41 @@ impl Drop for LocalHistogram {
     }
 }
 
+/// `LocalHistogramVec` can only be access by a single thread, like `LocalHistogramVec`.
```
Comment (disksing) [inline src/histogram.rs]: like `LocalHistogram`?

**Comment by disksing** [comment]
LGTM.

**Pull request merged**